Add helpful error message, on PhpParser syntax error

danon opened the pull request:
Before:
![image](https://user-images.githubusercontent.com/13367735/93940786-66e71f80-fd2d-11ea-82ee-c77523de7b66.png)

After:
![image](https://user-images.githubusercontent.com/13367735/93940792-6b133d00-fd2d-11ea-9086-176b68b91939.png)

danon committed:
Add helpful error message, on PhpParser syntax error

kelunik commented on bin/fqn-check:
```diff
@@ -41,20 +41,24 @@ $fileCount = count($files);
+        }
+    } catch (PhpParser\Error $e) {
+        print $e->getMessage() . " (errored in file: " . $file->getFilename() . ")\n";
     }
```
If we catch the exception, it needs to increase the error count, otherwise it exits as if everything was fine, no?

Danon replied:
I don't think so, because just because PhpParser didn't manage to parse the file, doesnt mean there are non-fully-qualified names in the file.

danon replied:
We can do something like: "There was 51 non-FQN files, and 6 files couldn't be parsed". Is it possible?

kelunik replied:
If you want to implement that, sure, go ahead! :-)

danon replied:
@kelunik Ok, I'll update this PR. In the meantime, you can take a look at the other one.

danon replied:
@kelunik Do you mind if I write unit tests in phpUnit?

kelunik replied:
Sure, go ahead. 😊

Danon replied:
But there's a problem, the script uses `exit()` to return status code, which terminates the unit tests :/

I would have to extract it to a separate file or a class.

Danon replied:
@kelunik

danon committed:
Use "new Lexer()" in fqn-check

danon committed:
Update .gitignore

danon committed:
Add PhpUnit, Add FqnCheckerTest unit test

Danon commented:
@kelunik I have written a test for FqnChecker, but to write test for the `/bin` script, I will have to move some of the logic to a class, if that's allright?

danon committed:
Install "vfsStream" dev dependency, Add scriptTest

danon committed:
Edit "fqn-check" (with exit()) to "fqn-check.php" (with return)

danon committed:
Add test cases to scriptTest.php

danon commented:
@kelunik Ok, I've written the tests, here's what I've done:

 - I moved the code from `fqn-check` to `fqn-check.php`, and replaced any `exit()` with `return ;`
 - In the original `fqn-check` I've left `exit(require 'fqn-check.php')`, so it will error code what was returned from `fqn-check-.php`
 - Thanks to that, I could've written unit tests (with `exit()` it was impossible, since the test framework would just quit).

You can run the tests just by calling
```
composer install
composer test
```
or from your IDE :)

danon committed:
Change project name to "rawr/fqn-check"

danon committed:
Change type to "library"

danon committed:
Update "nikic/php-parser" to v4.10.2

danon commented:
@kelunik Do you think it's okay now?

danon commented:
@kelunik Is this project still maintained?

kelunik commented:
@Danon Not really TBH, as you have noticed. We're using `php-cs-fixer`'s [`native_function_invocation`](https://github.com/amphp/php-cs-fixer-config/blob/3c24119d0377eed2093d5c0f0541478cb75ea72d/src/Config.php#L39) now.

I see that you already changed to a different package name. If you still maintain that, should I change the README to recommend your fork or `php-cs-fixer` instead?

danon commented:
@kelunik Well, I don't want to use `php-cs-fixer`, I like your solution with `fqn-check`. I use it in my project, and my friend also uses that.

I renamed the package because I thought you had abandoned the project.

If you don't mind I would like to maintain it. Would you feel conformable sharing the write access? Or should I continue on the fork?

kelunik commented:
Looking at the traffic here on GitHub and the install [statistics on Packagist](https://packagist.org/packages/kelunik/fqn-check/stats) it probably makes more sense to continue with the fork, as it already has more installs currently than this package. I'm happy to link it from the README if you don't mind. :smile:

danon commented:
> I'm happy to link it from the README if you don't mind. 😄

Sure, if you'd like :)

kelunik commented:
I've updated the README. I've removed the installation instructions to put more emphasis on the alternatives. I'll archive this repository now to make it read-only. :-)
